Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/PDFDataDetectorOverlayController.mm

```diff
@@ -237,15 +238,24 @@
     });
 }
 
-void PDFDataDetectorOverlayController::didInvalidateHighlightOverlayRects(ShouldUpdatePlatformHighlightData shouldUpdatePlatformHighlightData, ActiveHighlightChanged activeHighlightChanged)
+void PDFDataDetectorOverlayController::didInvalidateHighlightOverlayRects(std::optional<PDFDocumentLayout::PageIndex> pageIndex, ShouldUpdatePlatformHighlightData shouldUpdatePlatformHighlightData, ActiveHighlightChanged activeHighlightChanged)
 {
     // Regardless of what we repaint, we don't need the stale data after this.
     auto resetStaleDataDetectorWithHighlight = makeScopeExit([&] {
         m_staleDataDetectorItemWithHighlight = { { }, { } };
     });
 
     if (shouldUpdatePlatformHighlightData == ShouldUpdatePlatformHighlightData::Yes) {
-        WTF::forEach(m_pdfDataDetectorItemsWithHighlightsMap.keys(), [&](auto pageIndex) {
+        auto pageIndices = [&] {
+            if (pageIndex) {
```

Comment:
maybe this should be
```
if (!pageIndex) 
  return  m_pdfDataDetectorItemsWithHighlightsMap.keys();

// etc
```

to avoid extra nesting?

Reply:
I went with this first, but then settled for what exists currently because otherwise you need three `return` statements, and the tail looks awkward here.

```cpp
if (!pageIndex)
    return map.keys();

if (map.find(*pageIndex))
    return customRange(...pageIndex...);

return map.keys();
```